sdk/azcore/policy_body_download_test.go

```diff
@@ -234,10 +264,71 @@ func TestSkipBodyDownloadWith400(t *testing.T) {
 	if err != nil {
 		t.Fatalf("unexpected error: %v", err)
 	}
-	if len(resp.payload()) == 0 {
+	payload, err := resp.payload()
+	if err != nil {
+		t.Fatalf("unexpected error: %v", err)
+	}
+	if len(payload) == 0 {
 		t.Fatal("missing payload")
 	}
-	if string(resp.payload()) != message {
-		t.Fatalf("unexpected response: %s", string(resp.payload()))
+	if string(payload) != message {
+		t.Fatalf("unexpected response: %s", string(payload))
+	}
+}
+
+func TestReadBodyAfterSeek(t *testing.T) {
+	const message = "downloaded"
+	srv, close := mock.NewServer()
+	defer close()
+	srv.AppendResponse(mock.WithBody([]byte(message)))
+	srv.AppendResponse(mock.WithBody([]byte(message)))
+	// download policy is automatically added during pipeline construction
+	pl := NewPipeline(srv, NewRetryPolicy(testRetryOptions()))
+	req, err := NewRequest(context.Background(), http.MethodGet, srv.URL())
+	if err != nil {
+		t.Fatalf("unexpected error: %v", err)
+	}
+	resp, err := pl.Do(req)
+	if err != nil {
+		t.Fatalf("unexpected error: %v", err)
+	}
+	payload, err := resp.payload()
+	if err != nil {
+		t.Fatalf("unexpected error: %v", err)
+	}
+	if string(payload) != message {
+		t.Fatal("incorrect payload")
+	}
+	nb, ok := resp.Body.(*nopClosingBytesReader)
+	if !ok {
+		t.Fatalf("unexpected body type: %t", resp.Body)
+	}
+	_, err = nb.Seek(0, io.SeekStart)
+	if err != nil {
+		t.Fatalf("unexpected error: %v", err)
+	}
+	if nb.i != 0 {
```

Comment:
Instead of checking `nb.i` here, you should check against the returned offset from the call to `Seek()` above.  Comment applies to all invocations of `Seek()`.

Reply:
Done